### PR TITLE
feat: BankInfo step

### PR DESCRIPTION
**Pull request by MrMuzyk**
### Details

Updated BankInfo step for VBBA

### Fixed Issues

$ https://github.com/Expensify/App/issues/30244
PROPOSAL: 

### Tests

- [x] Verify that no errors appear in the JS console
- Enter adding bank account process for the workspace
- Select manual/plaid and add account

### QA Steps

- [x] Verify that no errors appear in the JS console
- Enter adding bank account process for the workspace
- Select manual/plaid and add account
- 
### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/23487612/f7ad1196-4242-4b9e-9c0c-467a971184ea

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/23487612/37b92d1c-ecd7-49e6-b4ad-451fbc3a83c8

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/23487612/8996a418-606b-4c28-aa89-29f06581e0cb

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/23487612/4c7f0e51-d45f-4c46-ab7c-cd78c4ddd00e

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/23487612/f4e0be87-8944-4966-8d68-502227e6465d

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/23487612/e20d8e95-1530-46a4-bf5e-dddf1d8b929a

</details>

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -234,8 +234,8 @@ PODS:
     - libwebp/demux
   - libwebp/webp (1.2.4)
   - lottie-ios (4.3.3)
-  - lottie-react-native (6.3.1):
-    - lottie-ios (~> 4.3.0)
+  - lottie-react-native (6.4.0):
```
Comment (MrMuzyk) [inline ios/Podfile.lock:237]: revert changes in this file

File: src/pages/ReimbursementAccount/BankInfo/substeps/Manual.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+const defaultProps = {
+    reimbursementAccount: ReimbursementAccountProps.reimbursementAccountDefaultProps,
+};
+
+const bankInfoStepKeys = CONST.BANK_ACCOUNT.BANK_INFO_STEP.INPUT_KEY;
+
+function Manual({reimbursementAccount, onNext}) {
+    const {translate, preferredLocale} = useLocalize();
+
+    const defaultValues = {
```
Comment (gedu) [inline src/pages/ReimbursementAccount/BankInfo/substeps/Manual.js:34]: does it make sense to wrap it in a `useMemo`?

Comment (MrMuzyk) [reply src/pages/ReimbursementAccount/BankInfo/substeps/Manual.js:34]: Honestly I dont think so. There are no expensive operations `perfomedgetDefaultValueForReimbursementAccountField` has only `lodashGet` that gets value for single field

File: src/components/AddPlaidBankAccount.js [hunk 3]
```diff
@@ -82,9 +87,19 @@ function AddPlaidBankAccount({
     bankAccountID,
     allowDebit,
     isPlaidDisabled,
+    isDisplayedInNewVBBA,
 }) {
+    const plaidBankAccounts = lodashGet(plaidData, 'bankAccounts') || [];
```
Comment (Swor71) [inline src/components/AddPlaidBankAccount.js:92]: shouldn't the default be within lodashGet?

File: src/components/AddPlaidBankAccount.js [hunk 4]
```diff
@@ -82,9 +87,19 @@ function AddPlaidBankAccount({
...
+    const defaultSelectedPlaidAccount = _.find(plaidBankAccounts, (account) => account.plaidAccountID === selectedPlaidAccountID);
+    const defaultSelectedPlaidAccountID = lodashGet(defaultSelectedPlaidAccount, 'plaidAccountID', '');
+    const defaultSelectedPlaidAccountMask = lodashGet(
+        _.find(plaidBankAccounts, (account) => account.plaidAccountID === selectedPlaidAccountID),
+        'mask',
+        '',
+    );
     const subscribedKeyboardShortcuts = useRef([]);
     const previousNetworkState = useRef();
+    const [selectedPlaidAccountMask, setSelectedPlaidAccountMask] = React.useState(defaultSelectedPlaidAccountMask);
```
Comment (Swor71) [inline src/components/AddPlaidBankAccount.js:102]: nit: add `useState` to already destructured imports

File: src/pages/ReimbursementAccount/BankInfo/substeps/Plaid.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            <AddPlaidBankAccount
+                onSelect={(plaidAccountID) => {
+                    ReimbursementAccount.updateReimbursementAccountDraft({plaidAccountID});
+                }}
+                plaidData={plaidData}
+                onExitPlaid={() => BankAccounts.setBankAccountSubStep(null)}
+                allowDebit
+                bankAccountID={bankAccountID}
+                selectedPlaidAccountID={selectedPlaidAccountID}
+                isDisplayedInNewVBBA
```
Comment (Swor71) [inline src/pages/ReimbursementAccount/BankInfo/substeps/Plaid.js:104]: will this come from somewhere else, or is it just a placeholder? the default value is set to `false`

File: src/components/AddPlaidBankAccount.js (hunk 3 above)
Comment (MrMuzyk) [reply src/components/AddPlaidBankAccount.js:92]: yes, Ive moved this line and didnt pay that much attention to it. Fixing it

File: src/pages/ReimbursementAccount/BankInfo/substeps/Plaid.js (hunk 5 above)
Comment (MrMuzyk) [reply src/pages/ReimbursementAccount/BankInfo/substeps/Plaid.js:104]: Everything is in place. This component is used somewhere esle in the codebase as well. This is the only place where we want to display updated UI inside of it thus prop set to true

File: src/pages/ReimbursementAccount/BankInfo/BankInfo.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    if (shouldReinitializePlaidLink) {
+        setupType = CONST.BANK_ACCOUNT.SETUP_TYPE.PLAID;
+    }
+
+    const startFrom = useMemo(() => getInitialSubstepForBankInfo(values, setupType), [setupType, values]);
+
+    const submit = useCallback(() => {
+        if (setupType === CONST.BANK_ACCOUNT.SETUP_TYPE.MANUAL) {
+            BankAccounts.connectBankAccountManually(
+                lodashGet(reimbursementAccount, 'achData', bankInfoStepKeys.BANK_ACCOUNT_ID) || 0,
```
Comment (Swor71) [inline src/pages/ReimbursementAccount/BankInfo/BankInfo.js:69]: please move the default value inside `lodashGet`

File: src/pages/ReimbursementAccount/BankInfo/BankInfo.js [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+                values[bankInfoStepKeys.ACCOUNT_NUMBER],
+                values[bankInfoStepKeys.ROUTING_NUMBER],
+                values[bankInfoStepKeys.PLAID_MASK],
+            );
+        } else if (setupType === CONST.BANK_ACCOUNT.SETUP_TYPE.PLAID) {
+            const bankAccountID = lodashGet(reimbursementAccount, 'achData.bankAccountID') || 0;
```
Comment (Swor71) [inline src/pages/ReimbursementAccount/BankInfo/BankInfo.js:75]: please move the default value inside `lodashGet`

File: src/pages/ReimbursementAccount/BankInfo/substeps/Plaid.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        const errorFields = {};
+        if (!values.acceptTerms) {
+            errorFields.acceptTerms = 'common.error.acceptTerms';
+        }
+
+        return errorFields;
+    }, []);
+
+    useEffect(() => {
+        const plaidBankAccounts = lodashGet(plaidData, 'bankAccounts') || [];
```
Comment (Swor71) [inline src/pages/ReimbursementAccount/BankInfo/substeps/Plaid.js:56]: please move the default value inside `lodashGet`

File: src/pages/ReimbursementAccount/BankInfo/substeps/Plaid.js [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+        const selectedPlaidBankAccount = _.findWhere(lodashGet(plaidData, 'bankAccounts', []), {
+            plaidAccountID: lodashGet(reimbursementAccountDraft, 'plaidAccountID', ''),
+        });
+
+        const bankAccountData = {
+            [bankInfoStepKeys.ROUTING_NUMBER]: selectedPlaidBankAccount.routingNumber,
+            [bankInfoStepKeys.ACCOUNT_NUMBER]: selectedPlaidBankAccount.accountNumber,
+            [bankInfoStepKeys.PLAID_MASK]: selectedPlaidBankAccount.mask,
+            [bankInfoStepKeys.IS_SAVINGS]: selectedPlaidBankAccount.isSavings,
+            [bankInfoStepKeys.BANK_NAME]: lodashGet(plaidData, 'bankName') || '',
```
Comment (Swor71) [inline src/pages/ReimbursementAccount/BankInfo/substeps/Plaid.js:73]: please move the default value inside `lodashGet`

File: src/pages/ReimbursementAccount/BankInfo/substeps/Plaid.js [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+            [bankInfoStepKeys.PLAID_ACCOUNT_ID]: selectedPlaidBankAccount.plaidAccountID,
+            [bankInfoStepKeys.PLAID_ACCESS_TOKEN]: lodashGet(plaidData, 'plaidAccessToken') || '',
```
Comment (Swor71) [inline src/pages/ReimbursementAccount/BankInfo/substeps/Plaid.js:75]: please move the default value inside `lodashGet`

File: src/pages/ReimbursementAccount/BankInfo/substeps/Plaid.js [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+        };
+
+        ReimbursementAccount.updateReimbursementAccountDraft(bankAccountData);
+        onNext();
+    }, [plaidData, reimbursementAccountDraft, onNext]);
+
+    const bankAccountID = getDefaultValueForReimbursementAccountField(reimbursementAccount, bankInfoStepKeys.BANK_ACCOUNT_ID) || 0;
```
Comment (Swor71) [inline src/pages/ReimbursementAccount/BankInfo/substeps/Plaid.js:82]: the default value should be the third argument of that fn

File: src/pages/ReimbursementAccount/utils/getInitialSubstepForPersonalInfo.js [hunk 12]
```diff
@@ -8,23 +8,23 @@ import CONST from '@src/CONST';
  * @returns {number}
  */
 function getInitialSubstepForPersonalInfo(data) {
-    if (data[CONST.BANK_ACCOUNT.PERSONAL_INFO_STEP.INPUT_KEY.FIRST_NAME] === undefined && data[CONST.BANK_ACCOUNT.PERSONAL_INFO_STEP.INPUT_KEY.LAST_NAME] === undefined) {
+    if (data[CONST.BANK_ACCOUNT.PERSONAL_INFO_STEP.INPUT_KEY.FIRST_NAME].length === 0 && data[CONST.BANK_ACCOUNT.PERSONAL_INFO_STEP.INPUT_KEY.LAST_NAME].length === 0) {
```
Comment (Swor71) [inline src/pages/ReimbursementAccount/utils/getInitialSubstepForPersonalInfo.js:11]: maybe just check against `''`?

File: src/pages/ReimbursementAccount/utils/getSubstepValues.js [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * Returns values for substep confirmation page
+ *
+ * @param {Object} inputKeys object that stores substep info keys
+ *
+ * @param {Object} reimbursementAccountDraft object that stores substep info draft data
+ *
+ * @param {Object} reimbursementAccount object that stores substep info data
+ *
+ * @returns {Object}
```
Comment (shubham1206agra) [inline src/pages/ReimbursementAccount/utils/getSubstepValues.js:14]:
```suggestion
 * @param {Object} inputKeys object that stores substep info keys
 * @param {Object} reimbursementAccountDraft object that stores substep info draft data
 * @param {Object} reimbursementAccount object that stores substep info data
 * @returns {Object}
```

File: src/pages/ReimbursementAccount/utils/getInitialSubstepForBankInfo.js [hunk 14]
```diff
@@ -0,0 +1,30 @@
+import CONST from '@src/CONST';
+
+/**
+ * Returns the initial substep for the Bank Info step based on already existing data
+ *
+ * @param {Object} data object that stores bank info data
+ *
+ * @param {string} setupType type of setup
+ *
+ * @returns {number}
```
Comment (shubham1206agra) [inline src/pages/ReimbursementAccount/utils/getInitialSubstepForBankInfo.js:10]:
```suggestion
 * @param {Object} data object that stores bank info data
 * @param {string} setupType type of setup
 * @returns {number}
```

File: src/pages/ReimbursementAccount/utils/getDefaultValueForReimbursementAccountField.js [hunk 15]
```diff
@@ -0,0 +1,14 @@
+import lodashGet from 'lodash/get';
+
+/**
+ * @param {Object} reimbursementAccount
+ * @param {String} fieldName
+ * @param {* | undefined} defaultValue
+ *
+ * @returns {String}
```
Comment (shubham1206agra) [inline src/pages/ReimbursementAccount/utils/getDefaultValueForReimbursementAccountField.js:8]:
```suggestion
 * @param {* | undefined} defaultValue
 * @returns {String}
```

**Comment by shubham1206agra** [comment]
Please transition this code to RNW v19

**Comment by MrMuzyk** [comment]
> Please transition this code to RNW v19

what exactly do you mean by this?

**Comment by shubham1206agra** [comment]
I think you need to fix your mWeb flow. It may show some console errors

**Comment by MrMuzyk** [comment]
> I think you need to fix your mWeb flow. It may show some console errors

can you add a log/video?

**Comment by gedu** [comment]
@shubham1206agra can you please provide steps, screens so we can understand better what do you mean? so we can address this faster, thanks

File: src/pages/ReimbursementAccount/BankInfo/substeps/Manual.js [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+                style={[styles.exampleCheckImage, styles.mb5]}
+                source={exampleCheckImage(preferredLocale)}
+            />
+            <TextInput
+                autoFocus
+                shouldDelayFocus={shouldDelayFocus}
+                inputID="routingNumber"
+                label={translate('bankAccount.routingNumber')}
+                accessibilityLabel={translate('bankAccount.routingNumber')}
+                accessibilityRole={CONST.ACCESSIBILITY_ROLE.TEXT}
```
Comment (shubham1206agra) [inline src/pages/ReimbursementAccount/BankInfo/substeps/Manual.js:90]:
```suggestion
                aria-label={translate('bankAccount.routingNumber')}
                role={CONST.ACCESSIBILITY_ROLE.TEXT}
```

File: src/pages/ReimbursementAccount/BankInfo/substeps/Manual.js [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+                defaultValue={defaultValues[bankInfoStepKeys.ROUTING_NUMBER]}
+                keyboardType={CONST.KEYBOARD_TYPE.NUMBER_PAD}
```
Comment (shubham1206agra) [inline src/pages/ReimbursementAccount/BankInfo/substeps/Manual.js:92]:
```suggestion
                inputMode={CONST.INPUT_MODE.NUMERIC}
```

File: src/pages/ReimbursementAccount/BankInfo/substeps/Manual.js [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+                disabled={shouldDisableInputs}
+                shouldSaveDraft
+                shouldUseDefaultValue={shouldDisableInputs}
+            />
+            <TextInput
+                inputID="accountNumber"
+                containerStyles={[styles.mt4]}
+                label={translate('bankAccount.accountNumber')}
+                accessibilityLabel={translate('bankAccount.accountNumber')}
+                accessibilityRole={CONST.ACCESSIBILITY_ROLE.TEXT}
```
Comment (shubham1206agra) [inline src/pages/ReimbursementAccount/BankInfo/substeps/Manual.js:102]:
```suggestion
                aria-label={translate('bankAccount.accountNumber')}
                role={CONST.ACCESSIBILITY_ROLE.TEXT}
```

**Comment by shubham1206agra** [comment]
@MrMuzyk Please see the suggestions

**Comment by MrMuzyk** [comment]
@shubham1206agra 
I will change to inputMode prop.

Why do you want me to change other props? According to eslint everything is fine. There are plenty of places in the code where it is done in the same way. What is wrong with it?

**Comment by shubham1206agra** [comment]
> @shubham1206agra I will change to inputMode prop.
> 
> Why do you want me to change other props? According to eslint everything is fine. There are plenty of places in the code where it is done in the same way. What is wrong with it?

Mainly accessibility props need to change due `react-native-web`

**Comment by MrMuzyk** [comment]
I

> > @shubham1206agra I will change to inputMode prop.
> > Why do you want me to change other props? According to eslint everything is fine. There are plenty of places in the code where it is done in the same way. What is wrong with it?
> 
> Mainly accessibility props need to change due `react-native-web`

Why isnt this enforced by eslint tho?

**Comment by shubham1206agra** [comment]
> > > @shubham1206agra I will change to inputMode prop.
> > > Why do you want me to change other props? According to eslint everything is fine. There are plenty of places in the code where it is done in the same way. What is wrong with it?
> > 
> > 
> > Mainly accessibility props need to change due `react-native-web`
> 
> Why isnt this enforced by eslint tho?

I don't know exactly

**Comment by MrMuzyk** [comment]
@shubham1206agra 
Fixed accessibility stuff.
What about this error on mWeb? Can you tell me a bit more what exactly is the problem?

**Comment by shubham1206agra** [comment]
See this PR https://github.com/Expensify/App/pull/24482

**Comment by shubham1206agra** [comment]
I will start the review after https://github.com/Expensify/App/pull/30924.
This PR will be reviewed by today/tomorrow morning.

**Comment by shubham1206agra** [comment]
Screenshot 2023-11-14 at 8 35 59 PM

**Comment by shubham1206agra** [comment]
@MrMuzyk Please fix the conflicts

**Comment by MrMuzyk** [comment]
> Screenshot 2023-11-14 at 8 32 40 PM
> This looks different from doc

We dont have round icons of banks so it will look slightly different

**Comment by shubham1206agra** [comment]
> > Screenshot 2023-11-14 at 8 32 40 PM
> > This looks different from doc
> 
> We dont have round icons of banks so it will look slightly different

I think checkbox is a little smaller

**Comment by shubham1206agra** [comment]
@MrMuzyk Can't we get icon for the bank?

**Comment by MrMuzyk** [comment]
> @MrMuzyk Can't we get icon for the bank?

Ive mentioned this on slack nearly 2 weeks ago. Got no response. Id treat it as possible improvement later on
image

**Comment by shubham1206agra** [comment]
https://github.com/Expensify/App/assets/58412969/6b284df0-4ba6-4e77-bf4a-60015d63be3a

@MrMuzyk I am facing this problem.

**Comment by MrMuzyk** [comment]
i'll have a look

**Comment by MrMuzyk** [comment]
@shubham1206agra Chase doesnt work on our feature branch as well. BE returns incorrect response for it. Its not a problem with my code.

image
image

https://github.com/Expensify/App/assets/23487612/53cd5c9f-1e02-4724-9192-9a4b4b76e3cd

**Comment by shubham1206agra** [comment]
> @shubham1206agra Chase doesnt work on our feature branch as well. BE returns incorrect response for it. Its not a problem with my code.
> image
> Screen.Recording.2023-11-15.at.13.12.24.mov

Getting to infinite loading on main
@mountiny Can you confirm this?

**Comment by MrMuzyk** [comment]
@shubham1206agra please check again, Ive added fix for the edge case

**Comment by MrMuzyk** [comment]
Where did you get this text from?

**Comment by shubham1206agra** [comment]
`plaidBodyCopy` in `en.ts`

**Comment by MrMuzyk** [comment]
But where is it metnioned that it has to be there?

**Comment by shubham1206agra** [comment]
Screenshot 2023-11-16 at 4 58 43 PM

**Comment by MrMuzyk** [comment]
Back button is disabled temporarily. Page wasnt behaving properly and this logic would get changed anyway. Once every step has its separate route then we will just navigate.goBack() there. Im working on ReimbursementAccountPage and Ive already implemented this change there. Can we agree on skipping it now and verifying it in next PR?

**Comment by MrMuzyk** [comment]
> Screenshot 2023-11-16 at 4 58 43 PM
> Maybe here (not sure)

ahh, so in the current flow it is there. Design has a dummy text in this place so in the end I didnt put anything in there
image

If you think it should be there I can add it

**Comment by shubham1206agra** [comment]
Lets add it there. And if there's any change, we will remove it there.

**Comment by MrMuzyk** [comment]
Added ^

**Comment by shubham1206agra** [comment]
> @shubham1206agra Is there everything else ok?

Looks fine. We will do improvement if there's any.

**Comment by MrMuzyk** [comment]
Awesome. Any chance you can do the checklist today so we can merge it?

**Comment by shubham1206agra** [comment]
https://github.com/Expensify/App/assets/58412969/766cccec-1235-494a-ad52-da5ba8241944

@MrMuzyk Can you check this at the end?

**Comment by MrMuzyk** [comment]
@shubham1206agra What about it exactly?

**Comment by shubham1206agra** [comment]
After the plaid connection, it shows the landing page for a brief time.

**Comment by MrMuzyk** [comment]
> After the plaid connection, it shows the landing page for a brief time.

It will not happen when this whole step will be on separate route. This is coming later on

**Comment by shubham1206agra** [comment]
One minor thing the next button in the manual step is getting hidden behind the keyboard but not on native. Can you check this?

**Comment by MrMuzyk** [comment]
> One minor thing the next button in the manual step is getting hidden behind the keyboard but not on native. Can you check this?

yes, checking

**Comment by shubham1206agra** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/58412969/9ea4c93e-1972-47e6-a4dc-5394ac4af1f5

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/58412969/e9606221-ccfb-4949-b4bd-5982da50a5fe

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/58412969/aa4e459b-7469-4662-9dc1-acdb435c555a

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/58412969/c716c3df-5aa8-451f-b5ad-b7cfc3f280fd

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/58412969/a0e99d1b-9ead-497b-ad5a-e275f209b53b

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/58412969/db93403b-c8f3-46b9-b6fc-81f9c89671cf

</details>

**Comment by MrMuzyk** [comment]
image
I think it looks fine for me doesnt it?

**Comment by shubham1206agra** [comment]
> image I think it looks fine for me doesnt it?

It looks fine in this. But see video in my checklist for mWeb

**Comment by mountiny** [comment]
Let me know in slack if this will be ready to merge

**Comment by MrMuzyk** [comment]
@shubham1206agra 

https://github.com/Expensify/App/assets/23487612/11c9f3dc-3391-46dc-bd77-e60403b8c4ab

**Comment by shubham1206agra** [comment]
Is it not related to this PR then?

**Comment by MrMuzyk** [comment]
@shubham1206agra 
Im not sure what to do about it honestly. Im using regular components like in previous steps. Im looking at what is causing this but I have no clue tbh. Is it a blocker? The button is visible once user gets to last field and can submit

**Comment by MrMuzyk** [comment]
image
TS checks are failing because of new JS files

**Review by shubham1206agra — APPROVED** [review]

**Review by mountiny — APPROVED** [review]
Thanks lets keep going

**Comment by mountiny** [comment]
Thsi goes to a feature branch so failing ts check can be handled later

**Pull request merged**